packages/peregrine/lib/talons/Header/useCartTrigger.js

```diff
@@ -1,24 +1,38 @@
 import { useCallback, useEffect } from 'react';
-import { useMutation } from '@apollo/react-hooks';
+import { useLazyQuery, useMutation } from '@apollo/react-hooks';
 import { useCartContext } from '@magento/peregrine/lib/context/cart';
 import { useAppContext } from '@magento/peregrine/lib/context/app';
 import { useAwaitQuery } from '@magento/peregrine/lib/hooks/useAwaitQuery';
 
 export const useCartTrigger = props => {
-    const { createCartMutation, getCartDetailsQuery } = props;
+    const {
+        mutations: { createCartMutation },
+        queries: { getCartDetailsQuery, getItemCountQuery }
+    } = props;
     const [, { toggleDrawer }] = useAppContext();
-    const [{ derivedDetails }, { getCartDetails }] = useCartContext();
-    const { numItems: itemCount } = derivedDetails;
+    const [{ cartId }, { createCart, getCartDetails }] = useCartContext();
 
+    const [getItemCount, { data }] = useLazyQuery(getItemCountQuery);
     const [fetchCartId] = useMutation(createCartMutation);
     const fetchCartDetails = useAwaitQuery(getCartDetailsQuery);
 
+    const itemCount = data ? data.cart.total_quantity : 0;
+
     useEffect(() => {
-        getCartDetails({
-            fetchCartId,
-            fetchCartDetails
-        });
-    }, [fetchCartDetails, fetchCartId, getCartDetails]);
+        if (!cartId) {
```

Comment:
Previously this happened somewhat "magically", by calling the `getCartDetails` action, which then created a cart if one was not present. I just made it more explicit w/o mucking much with the redux portion of this logic.

Reply:
I tried this in a spike but I kept it at the root level like in `useCartContext`. It's one of those core assumptions of using the cart context - that you'll always have a valid cart, so I felt like this effect here should just come bundled with the context hook. We should avoid having cart creation be a part of each component that wants to use a cart or we will end up with multiple mutation requests for the id...which will be super slow now that we queue them all 😨

Reply:
Discussed some more on Slack, and I've decided to revert this change. We'll limit the scope of this PR and tackle larger architecture questions surrounding cart creation and retry logic in a future Story.